src/plugins/console/server/routes/api/console/proxy/create_handler.ts

```diff
@@ -124,13 +66,28 @@ function getProxyHeaders(req: OpenSearchDashboardsRequest) {
   return headers;
 }
 
+function toUrlPath(path: string) {
+  let urlPath = `/${trimStart(path, '/')}`;
+  // Appending pretty here to have OpenSearch do the JSON formatting, as doing
+  // in JS can lead to data loss (7.0 will get munged into 7, thus losing indication of
+  // measurement precision)
+  if (!urlPath.includes('?pretty')) {
+    urlPath += '?pretty=true';
+  }
+  return urlPath;
+}
+
 export const createHandler = ({
   log,
   proxy: { readLegacyOpenSearchConfig, pathFilters, proxyConfigCollection },
 }: RouteDependencies): RequestHandler<unknown, Query, Body> => async (ctx, request, response) => {
   const { body, query } = request;
   const { path, method } = query;
 
+  let opensearchResponse: ApiResponse;
+
+  const client = ctx.core.opensearch.client.asCurrentUser;
```

Comment:
nit: should we move this `const client` to line 86?  then `let opensearchResponse` to line 88?